File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        res = [1] * len(nums)
        prev = 1
        after = 1
        for i in range(len(nums)):
            if i > 0:
                prev *= nums[i-1]
            else:
                prev*=1
            res[i] *= prev
        for i in range(len(nums)-1,-1,-1):
            if i < len(nums)-1:
                after *= nums[i+1]
            else:
                after *=1
            res[i] *= after
        return res

File: Data Structures & Algorithms/subsets/submission-0.js
class Solution {
    path = []
    res = []
    /**
     * @param {number[]} nums
     * @return {number[][]}
     */
    subsets(nums) {
        this.dfs(nums,0)
        return this.res;
    }
    dfs(nums,i){
        if(i === nums.length){
            this.res.push([...this.path])
            return
        }
        this.path.push(nums[i])
        this.dfs(nums, i + 1)

        this.path.pop()
        this.dfs(nums, i + 1)
    }

}
